chia/wallet/wallet_state_manager.py

```diff
@@ -965,15 +962,15 @@ async def coin_added(
                 sent_to=[],
                 trade_id=None,
                 type=uint32(tx_type),
-                name=coin.name(),
+                name=coin_name,
                 memos=[],
             )
             await self.tx_store.add_transaction_record(tx_record, True)
         else:
             records: List[TransactionRecord] = []
-            for record in all_outgoing_transaction_records:
+            for record in all_unconfirmed_transaction_records:
                 for add_coin in record.additions:
-                    if add_coin.name() == coin.name():
+                    if add_coin == coin:
                         records.append(record)
```

Comment:
one more question about this pattern. We have this block, essentially, in 4 places. What I find a bit odd is that this code seems to expect that `coin` (or `coin_state.coin` in other instances) can occur multiple times in `record.additions` / `out_tx_record.removals`.

I would be surprised if they can. Once we find the coin, why do we keep searching for it?

This boils down to that these loops, at least the inner loop, should probably be changed to an actual search call. e.g.
```
for record in all_outgoing_transaction_records:
    if coin in record.additions:
        records.append(record)
        break
```